src/System.Private.CoreLib/src/System/Threading/Thread.CoreRT.Windows.cs

```diff
@@ -319,23 +334,33 @@ internal static void InitializeCom(ApartmentState state = ApartmentState.MTA)
 
         private static void UninitializeCom()
         {
-            if (!t_comInitializedByUs)
+            if ((t_comState & ComState.InitializedByUs) == 0)
                 return;
 
 #if ENABLE_WINRT
             Interop.WinRT.RoUninitialize();
 #else
             Interop.Ole32.CoUninitialize();
 #endif
-            t_comInitializedByUs = false;
+
+            t_comState &= ~ComState.InitializedByUs;
         }
 
         // TODO: https://github.com/dotnet/corefx/issues/20766
         public void DisableComObjectEagerCleanup() { }
 
         private static void InitializeExistingThreadPoolThread()
         {
+#if ENABLE_WINRT
```

Comment:
I am not sure how to test whether this works well for WinRT, so I have left it behave exactly as before.

Reply:
👍 sounds good!